src/main/java/com/minecolonies/coremod/commands/generalcommands/ScanCommand.java

```diff
@@ -0,0 +1,117 @@
+package com.minecolonies.coremod.commands.generalcommands;
+
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.minecolonies.coremod.commands.AbstractSingleCommand;
+import com.minecolonies.coremod.commands.ActionMenu;
+import com.minecolonies.coremod.commands.IActionCommand;
+import com.minecolonies.coremod.items.ItemScanTool;
+
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.World;
+
+/**
+ * Created by asie on 2/16/17.
+ */
+public class ScanCommand extends AbstractSingleCommand implements IActionCommand
+{
+    public static final String DESC                   = "scan";
+    public static final String NO_PERMISSION_MESSAGE  = "You do not have permission to scan structures!";
+    public static final String SCAN_SUCCESS_MESSAGE = "Successfully scan structure!";
+    public static final String SCAN_FAILURE_MESSAGE = "Failed to scan structure!";
+
+    /**
+     * no-args constructor called by new CommandEntryPoint executer.
+     */
+    public ScanCommand()
+    {
+        super();
+    }
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public ScanCommand(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
+    {
+        // Will throw ClassCastException if null, but should never be null as these values are required.
+        final int x1 = actionMenu.getIntegerForArgument("x1");
+        final int y1 = actionMenu.getIntegerForArgument("y1");
+        final int z1 = actionMenu.getIntegerForArgument("z1");
+        final int x2 = actionMenu.getIntegerForArgument("x2");
+        final int y2 = actionMenu.getIntegerForArgument("y2");
+        final int z2 = actionMenu.getIntegerForArgument("z2");
+        final BlockPos from = new BlockPos(x1, y1, z1);
+        final BlockPos to = new BlockPos(x2, y2, z2);
+        executeShared(server, sender, from, to);
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+//        executeShared(server, sender, from, to);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)